website_sale_wishlist/models/product_wishlist.py

```diff
@@ -68,10 +68,12 @@ def _join_current_user_and_session(self):
         user_products = self.search([
             ("user_id", "=", self.env.uid),
         ]).mapped("product_tmpl_id")
-        session_domain = [
-            ("session", "=", request.session.sid),
-            ("user_id", "=", False),
-        ]
+
+        if request:
```

Comment:
I agree with @Yajo - this method should just be skipped with no session, because there's no possible wishlist.

IMO this should be changed back & the following should be added to top of this method:

```
if not request:
    return
```